test/mask_test.py

```diff
@@ -452,10 +458,167 @@ def test_from_threshold(self):
             self.assertEqual(mask.count(), 100)
             self.assertEqual(mask.get_bounding_rects(), [pygame.Rect((40,40,10,10))])
 
+    def test_zero_mask(self):
+        mask = pygame.mask.Mask((0, 0))
+        self.assertEqual(mask.get_size(), (0, 0))
 
+        mask = pygame.mask.Mask((100, 0))
+        self.assertEqual(mask.get_size(), (100, 0))
+
+        mask = pygame.mask.Mask((0, 100))
+        self.assertEqual(mask.get_size(), (0, 100))
 
-if __name__ == '__main__':
+    def test_zero_mask_overlap(self):
+        sizes = ((100, 0), (0, 100), (0, 0))
+
+        for size in sizes:
+            mask = pygame.mask.Mask(size)
+            mask2 = pygame.mask.Mask((100, 100))
+            self.assertEqual(mask.overlap(mask2, (0, 0)), None)
+            self.assertEqual(mask2.overlap(mask, (0, 0)), None)
+
+    def test_zero_mask_overlap_area(self):
+        sizes = ((100, 0), (0, 100), (0, 0))
+
+        for size in sizes:
+            mask = pygame.mask.Mask(size)
+            mask2 = pygame.mask.Mask((100, 100))
+            self.assertEqual(mask.overlap_area(mask2, (0, 0)), 0)
+            self.assertEqual(mask2.overlap_area(mask, (0, 0)), 0)
+
+
+    def test_zero_mask_overlap_mask(self):
+        sizes = ((100, 0), (0, 100), (0, 0))
+
+        for size in sizes:
+            mask = pygame.mask.Mask(size)
+            mask2 = pygame.mask.Mask((100, 100))
+
+            overlap_mask = mask.overlap_mask(mask2, (0, 0))
+            overlap_mask2 = mask2.overlap_mask(mask, (0, 0))
+
+            self.assertEqual(mask.get_size(), overlap_mask.get_size())
+            self.assertEqual(mask2.get_size(), overlap_mask2.get_size())
+
+    def test_zero_mask_fill(self):
+        sizes = ((100, 0), (0, 100), (0, 0))
+
+        for size in sizes:
+            mask = pygame.mask.Mask(size)
+            mask.fill()
+            self.assertEqual(mask.count(), 0)
+
+    def test_zero_mask_clear(self):
+        sizes = ((100, 0), (0, 100), (0, 0))
+
+        for size in sizes:
+            mask = pygame.mask.Mask(size)
+            mask.clear()
+            self.assertEqual(mask.count(), 0)
+
+    def test_zero_mask_flip(self):
+        sizes = ((100, 0), (0, 100), (0, 0))
+
+        for size in sizes:
+            mask = pygame.mask.Mask(size)
+            mask.invert()
+            self.assertEqual(mask.count(), 0)
 
+    def test_zero_mask_scale(self):
+        sizes = ((100, 0), (0, 100), (0, 0))
+
+        for size in sizes:
+            mask = pygame.mask.Mask(size)
+            mask.scale((2, 3))
```

Comment:
mask.scale should return a new mask with the requested size.
https://www.pygame.org/docs/ref/mask.html#pygame.mask.Mask.scale

So the test should be something like...
```python
scaled_mask = mask.scale((2, 3))
self.assertEqual(scaled_mask.get_size(), (2, 3))
```